packages/create-amplify/src/package-manager-controller/package_manager_controller.ts

```diff
@@ -0,0 +1,108 @@
+import { existsSync as _existsSync } from 'fs';
+import { execa as _execa } from 'execa';
+import * as path from 'path';
+import { logger } from '../logger.js';
+import { executeWithDebugLogger as _executeWithDebugLogger } from '../execute_with_logger.js';
+import { InitialProjectFileGenerator } from '../initial_project_file_generator.js';
+import { DependencyType } from './package_manager_controller_factory.js';
+
+/**
+ *
+ */
+export class PackageManagerController {
+  protected executable: string;
+  protected binaryRunner: string;
+  protected initDefault: string[];
+  protected installCommand: string;
+  protected readonly execa = _execa;
+
+  private readonly executeWithDebugLogger = _executeWithDebugLogger;
+  private readonly existsSync = _existsSync;
+
+  /**
+   * constructor
+   */
+  constructor(readonly projectRoot: string) {}
+
+  /**
+   * Installs the given package names as devDependencies
+   */
+  installDependencies = async (
+    packageNames: string[],
+    type: DependencyType
+  ): Promise<void> => {
+    const args = [`${this.installCommand}`].concat(...packageNames);
+    if (type === 'dev') {
+      args.push('-D');
+    }
+
+    await this.executeWithDebugLogger(
+      this.projectRoot,
+      this.executable,
+      args,
+      this.execa
+    );
+  };
+
+  getWelcomeMessage = () => {
+    const cdCommand =
+      process.cwd() === this.projectRoot
+        ? ''
+        : `cd .${this.projectRoot.replace(process.cwd(), '')}; `;
+
+    return `Welcome to AWS Amplify! 
+Run \`${this.binaryRunner} amplify help\` for a list of available commands. 
+Get started by running \`${cdCommand}${this.binaryRunner} amplify sandbox\`.`;
+  };
+
+  /**
+   * generateInitialProjectFiles
+   */
+  async generateInitialProjectFiles() {
```

Comment:
Oh, looks like the linter is turned off for this branch? We should reenable that. Otherwise, it's just going to be a bunch of failures when merging back to main.

Reply:
let me try merging `main` to the branch 🤞

Reply:
Done b19a209656337d5fe39b18835ab767fc1a11f9e5